modules/global/src/com/haulmont/cuba/core/sys/AbstractViewRepository.java

```diff
@@ -438,9 +457,17 @@ protected View deployView(Element rootElem, Element viewElem, Set<ViewInfo> visi
         View v = retrieveView(metaClass, viewName, visited);
         boolean overwrite = Boolean.parseBoolean(viewElem.attributeValue("overwrite"));
 
-        String ancestor = viewElem.attributeValue("extends");
-        if (!overwrite) {
-            overwrite = StringUtils.equals(ancestor, viewName);
+        String extended = viewElem.attributeValue("extends");
+        List<String> ancestors = null;
+
+        if (isNotBlank(extended)) {
+            ancestors = splitExtends(extended);
+        }
+
+        if (!overwrite && ancestors != null) {
+            overwrite = ancestors.stream().anyMatch(

```

Comment:
Again, can be replaced with `list.contains()`

Reply:
done